sklearn/utils/estimator_checks.py

```diff
@@ -131,6 +131,29 @@ def _yield_classifier_checks(name, Classifier):
     if 'class_weight' in Classifier().get_params().keys():
         yield check_class_weight_classifiers
 
+def check_supervised_y_no_nan(name, Estimator):
+    # Checks that the Estimator targets are not NaN.
+
+    warnings.simplefilter("ignore")
+    X = np.random.randn(10, 5)
+    y1 = np.random.randn(10) / 0.
+    y2 = np.random.randn(10, 2) / 0.
+
+    errmsg = "Input contains NaN, infinity or a value too large for " \
+             "dtype('float64')."
+    try:
+        if "MultiTask" in name:
+            Estimator().fit(X, y2)
```

Comment:
Whoops. Nice catch. 

However I still think code reuse would be a good idea.

Can you

1. change `if name in ([])` to `if "MultiTask" in name`
2. `return y[:, np.newaxis]` to `return np.reshape(y, (-1, 1))`